src/source.rs

```diff
@@ -0,0 +1,118 @@
+use anyhow::{anyhow, Result};
+use crossbeam_channel::{Receiver, Sender};
+use std::{path::PathBuf, thread};
+
+use compositor_common::{scene::Resolution, Frame};
+use compositor_pipeline::pipeline::PipelineSource;
+use ffmpeg_next::{
+    codec::{self, Context, Id},
+    format::{self, Pixel},
+    frame, Dictionary, Rational,
+};
+
+pub struct RtpSource {
+    #[allow(dead_code)]
+    port: u16,
+    sender: Sender<Frame>,
+}
+
+impl RtpSource {
+    pub fn new(port: u16, resolution: Resolution) -> Self {
+        let port_clone = port;
+        let (sender, receiver) = crossbeam_channel::unbounded();
+        thread::spawn(move || RtpSource::run(port_clone, resolution, receiver).unwrap());
+        Self { port, sender }
+    }
+
+    fn run(port: u16, resolution: Resolution, receiver: Receiver<Frame>) -> Result<()> {
+        let mut output_ctx =
+            format::output_as(&PathBuf::from(format!("rtp://127.0.0.1:{}", port)), "rtp")?;
+        let h264_codec = codec::encoder::find(Id::H264).unwrap();
+        let mut stream = output_ctx.add_stream(h264_codec)?;
+        unsafe {
+            (*(*stream.as_mut_ptr()).codecpar).codec_id = codec::Id::H264.into();
+        }
+
+        let mut encoder = Context::new().encoder().video()?;
+        encoder.set_time_base(Rational::new(1, 90000));
```

Comment:
I have no idea why we use `Rational::new(1, 90000)` here. 
I think it should look like this:
```rust
        let some_self_explaing_name = Rational::new(1, 90000);
        encoder.set_time_base(some_self_explaing_name);
```